std/http/file_server.ts

```diff
@@ -124,7 +124,7 @@ export async function serveFile(
   return new ServerResponse({
     body: file,
     headers: {
-      "content-type": contentType(filePath) ?? fileInfo.size.toString(),
+      "content-length": contentType(filePath) ?? fileInfo.size.toString(),
```

Comment:
The problem is that `contentType(filePath)` result should be the value of `content-type` header and `fileInfo.size.toString()` should be the value of `content-length` header. It is not possible to merge them here.

Reply:
I have no idea how I messed that one up... I'll fix it.
Should I fix the conflicts too?

Reply:
I think proper types for `headers` could help. But that's probably should be a separate PR.

Reply:
What do you mean?
Headers should match the interface `HeadersInit`

Reply:
I was thinking about something like this:

```ts
type KnownHeaders = {
  "content-type"?: string;
  "content-length"?: number;
  // other known headers
  [header: string]: string | string[] | number | undefined;
}

type HeadersInit = Headers | string[][] | Record<string, string> | KnownHeaders;
```

Does it make sense?

Reply:
Looks like it should work.
But if we want to add a list of know headers, then there is a long list... and headers are not case sensitive. So I don't know if it's a good idea.

The Fetch spec doesn't restrict the number of valid headers, but they mention these(is different contexts):
`Accept-Charset`
`Accept-Encoding`
`Access-Control-Request-Headers`
`Access-Control-Request-Method`
`Connection`
`Content-Length`
`Cookie`
`Cookie2`
`Date`
`DNT`
`Expect`
`Host`
`Keep-Alive`
`Origin`
`Referer`
`TE`
`Trailer`
`Transfer-Encoding`
`Upgrade`
`Via`
`Content-Encoding`
`Content-Language`
`Content-Location`
`Content-Type`
(there are probably more in the spec that I've missed)

Reply:
Sorry for late answer. It will not be restricted to some limited headers. It will only enforce the proper type if you use a predefined header. But anyway, the case insensitivity is a show stopper here :disappointed:. Never thought about it.